src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -828,4 +828,6 @@
      (let [{:keys [filter-groups selected-index data on-change]} props]
        (when selected-index
          (let [{:keys [pred]} (nth filter-groups selected-index)]
-           (on-change selected-index (if pred (filter pred data) data))))))})
+           (js/setTimeout
```

Comment:
This may benefit from a comment. I might suggest, "A user of this component might expect this event to fire only via user interaction long after the page has loaded, so add this call to the end of the event loop to better match this expectation."